Matrix.py

```diff
@@ -122,54 +125,44 @@ def matrices_mul(self, other):
         """multiply a matrix by matrix
                     args: self.data (tuple)- A matrix
                         other(tuple) -A matrix
-
                     returns: return the new matrix (tuple)"""
-        if type(other) is Matrix:  # Matrix's multiplication --->
-            answer = []  # will contain the answer as a list of lists.
-            new_other = []
-            # this loop convert "other" from Matrix type to a list of lists named "new_other".
-            for line in range(len(self.data)):
-                new_other.append([])
-                answer.append([0] * len(self.data))  # ns starts as a list of lists full of zeros.
-                for num in range(len(self.data)):
-                    new_other[line].append(other[num][line])
-            # this is the multiplication process->
-            for tup in range(len(self.data)):
-                for num in range(len(self.data)):
-                    for i in range(len(self.data)):
-                        answer[tup][num] += self.data[tup][i] * new_other[num][i]
-#                here the program adds to the zero the summery of the multiplications of
-#                any two numbers which their places in the matrices are fits by the matrices
-#                multiplication rules.
-            sub_list = [tuple(ls) for ls in answer]
-            # convert answer to type Matrix. ls are the inside lists.
-            return Matrix((tuple(sub_list)))
+        answer = []
+        new_other = []
+        # this loop convert "other" from Matrix type to a list of lists named "new_other".
+        for line in range(len(self.data)):
+            new_other.append([])
+            answer.append([0] * len(self.data))
+            for num in range(len(self.data)):
+                new_other[line].append(other[num][line])
+        # this is the multiplication process->
+        for line in range(len(self.data)):
+            for num in range(len(self.data)):
+                for i in range(len(self.data)):
+                    answer[line][num] += self.data[line][i] * new_other[num][i]
+#                   here the program adds to the zero the summery of the multiplications of
+#                   any two numbers which their places in the matrices are fits by the matrices
+#                   multiplication rules.
+        return Matrix((tuple([tuple(ls) for ls in answer])))
```

Comment:
again, don't use matrix

Reply:
are you sure? Daniel told me he also did it this way and i my tries to change it are not working

Reply:
I told Daniel the same thing